openshift_metrics/metrics_processor.py

```diff
@@ -22,9 +22,10 @@ def merge_metrics(self, metric_name, metric_list):
             gpu_resource = None
             node_model = None
 
-            unique_name = namespace + "+" + pod
-            if unique_name not in self.merged_data:
-                self.merged_data[unique_name] = {"namespace": namespace, "metrics": {}}
+            if namespace not in self.merged_data:
+                self.merged_data[namespace] = {}
+            if pod not in self.merged_data[namespace]:
+                self.merged_data[namespace][pod] = {"metrics": {}}
```

Comment:
This and the other lines in this file which check for a key before providing an initial value can be simplified with `setdefault()`. Docs [here](https://docs.python.org/3/library/stdtypes.html#dict.setdefault).  

```python
>>> a = {'a': []}
>>> a.setdefault('a', 'blah')
[]
>>> a.setdefault('b', 'blah')
'blah'
>>> a
{'a': [], 'b': 'blah'}
>>> 

```

Or/And, you can make use of a `defaultdict`, which provides an initial value for all keys which don't exist. Docs [here](https://docs.python.org/3/library/collections.html#collections.defaultdict). 

```python
>>> from collections import defaultdict
>>> b = defaultdict(dict)
>>> b['missing_key']
{}
>>> b
defaultdict(<class 'dict'>, {'missing_key': {}})
```

Or a combination of both of the above. (setdefault a defaultdict)

Reply:
that's useful. I updated it to use `setfedault` in this case.